upd readme [duda] (#1)
DevExpress-Examples/aspxgridview-batch-edit-cancel-editor-row-editing-in-the-client-focusedcellchanging-event-t496531: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T496531)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# ASP.NET - Handle the client-side FocusedCellChanging event to prevent an edit action for ASPxGridView cells

This example handles the client-side [`FocusedCellChanging`](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.FocusedCellChanging) event to prevent users from editing corresponding grid cells in [batch edit mode](https://docs.devexpress.com/AspNet/16443/components/grid-view/concepts/edit-data/batch-edit-mode).

<!-- run online -->
**[[Run Online]](https://codecentral.devexpress.com/t496531/)**
<!-- run online end -->

## Files to Look At

* [Default.aspx](./CS/Default.aspx) (VB: [Default.aspx](./VB/Default.aspx))
* [Default.aspx.cs](./CS/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/Default.aspx.vb))

## Implementation Details

> On the client side, the [`FocusedCellChanging`](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.FocusedCellChanging) event contains clicked cell data of the [`ASPxClientGridViewCellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewCellInfo) class. To get this data, the code below uses the event handler argument's [`cellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewFocusedCellChangingEventArgs.cellInfo) property.

Reviewer: ```suggestion
On the client side, the [`FocusedCellChanging`](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.FocusedCellChanging) event contains the [`cellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewFocusedCellChangingEventArgs.cellInfo) event argument that specifies the clicked cell's data (an [`ASPxClientGridViewCellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewCellInfo) object).
```

Author: ```suggestion
The following client-side [`FocusedCellChanging`] event handler uses the [`e.Cancel`] property to cancel the focus action (and subsequent edit operations) for cells in specific columns and rows. The code uses the [`e.cellInfo`] event property to get information on the clicked cell's column and row.
```

> Depending on the obtained data, the handler code determines whether a user can focus and then edit the clicked cell. For this purpose, the code specifies the handler argument's [`cancel`](https://docs.devexpress.com/AspNet/js-ASPxClientCancelEventArgs.cancel) property. When the [`cancel`](https://docs.devexpress.com/AspNet/js-ASPxClientCancelEventArgs.cancel) property is **true**, the grid cancels the focus action for the clicked cell.

Reviewer: ```suggestion
The event handler reads the cell's data to determine whether a user can focus and then edit the clicked cell. Set the [`cancel`](https://docs.devexpress.com/AspNet/js-ASPxClientCancelEventArgs.cancel) event property to **true** to cancel the focus action for the clicked cell.
```

Author: ```suggestion
```

}
```

> This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in the row switched to edit mode except for cells in read-only columns (the [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).

Reviewer 2: ```suggestion
This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in rows switched to edit mode except for cells in read-only columns (the [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).
```

Author: As a result of the discussion, we've decided to keep the "the row".

}
```

> Note that this technique does not work if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and then edit any cell in the row switched to edit mode (except for cells of columns with the enabled [`ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).

Reviewer: ```suggestion
This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in the row switched to edit mode except for read-only column cells (see [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly)).
```

Author: ```suggestion
This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in the row switched to edit mode except for cells in read-only columns (the [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).
```
